lib/Basic/LangOptions.cpp

```diff
@@ -199,6 +200,14 @@ checkPlatformCondition(PlatformConditionKind Kind, StringRef Value) const {
   if (Kind == PlatformConditionKind::OS && Value == "macOS")
     return checkPlatformCondition(Kind, "OSX");
 
+  if (Kind == PlatformConditionKind::OS && Value == "Darwin") {
+    // "Darwin" is an alias for all Darwin platforms.
+    return checkPlatformCondition(Kind, "OSX") ||
```

Comment:
LLVM has a predicate for this already [`Triple::isOSDarwin()`](https://llvm.org/doxygen/classllvm_1_1Triple.html#ab6fdf9b428bc3d57837022121c155cbf). Let's use that instead of a list of conditionals.

Reply:
How would we access that from the scope of this function?

Reply:
You have access to `Target`, which is the parsed value of -target, in this function.

Reply:
Should be good now